static/js/components/FindingChart.jsx

```diff
@@ -0,0 +1,280 @@
+import React, { useRef, useState, Suspense } from "react";
+import { useParams } from "react-router-dom";
+import { makeStyles } from "@material-ui/core/styles";
+
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+import InputLabel from "@material-ui/core/InputLabel";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import FormControl from "@material-ui/core/FormControl";
+import Button from "@material-ui/core/Button";
+import Checkbox from "@material-ui/core/Checkbox";
+import Input from "@material-ui/core/Input";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import Typography from "@material-ui/core/Typography";
+import PrintIcon from "@material-ui/icons/Print";
+
+import { useImage } from "react-image";
+import { useForm, Controller } from "react-hook-form";
+import { useReactToPrint } from "react-to-print";
+
+const useStyles = makeStyles((theme) => ({
+  button: {
+    margin: theme.spacing(1),
+  },
+  paper: {
+    width: "100%",
+    padding: theme.spacing(1),
+    textAlign: "left",
+    color: theme.palette.text.primary,
+  },
+  items: {
+    maxWidth: "100%",
+    width: "10rem",
+    fullWidth: "true",
+    display: "flex",
+    wrap: "nowrap",
+  },
+  spinner: {
+    position: "relative",
+    marginLeft: "50%",
+    width: "100%",
+  },
+}));
+
+const FindingChart = () => {
+  const classes = useStyles();
+  const { handleSubmit, getValues, errors, control } = useForm();
+  const { id } = useParams();
+
+  const [params, setParams] = useState({
+    imagesource: "desi",
+    useztfref: true,
+    findersize: 4.0,
+    howmany: 3,
+  });
+
+  const componentRef = useRef();
+
+  const initialFormState = {
+    ...params,
+  };
+
+  const url = [
```

Comment:
Would be cleaner to use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams